lib/improver/tests/lapse_rate/test_apply_lapse_rate.py

```diff
@@ -0,0 +1,168 @@
+                   self.temperature.data[source_lt_dest]))
+
+        # test that temperatures are increased where destination orography
+        # is lower than source
+        source_gt_dest = np.where(self.source_orog.data > self.dest_orog.data)
+        self.assertTrue(
+            np.all(result.data[source_gt_dest] >
+                   self.temperature.data[source_gt_dest]))
+
+        # test that temperatures are equal where destination orography
+        # is equal to source
+        source_eq_dest = np.where(np.isclose(self.source_orog.data,
+                                             self.dest_orog.data))
+        self.assertArrayAlmostEqual(result.data[source_eq_dest],
+                                    self.temperature.data[source_eq_dest])
+
+        # match specific values
+        self.assertArrayAlmostEqual(result.data, self.expected_data)
+
+    def test_unit_adjustment(self):
+        """Test correct values are retrieved if input cubes have incorrect
+        units"""
+        self.temperature.convert_units('degC')
+        self.source_orog.convert_units('km')
+        result = apply_lapse_rate(self.temperature, self.lapse_rate,
+                                  self.source_orog, self.dest_orog)
+        self.assertEqual(result.units, 'K')
+        self.assertArrayAlmostEqual(result.data, self.expected_data)
+
+    def test_realizations(self):
+        """Test processing of a cube with multiple realizations"""
+        temp_3d = add_coordinate(self.temperature, [0, 1, 2], 'realization')
+        lrt_3d = add_coordinate(self.lapse_rate, [0, 1, 2], 'realization')
+        result = apply_lapse_rate(
+            temp_3d, lrt_3d, self.source_orog, self.dest_orog)
+        self.assertArrayEqual(
+            result.coord('realization').points, np.array([0, 1, 2]))
+        for subcube in result.slices_over('realization'):
+            self.assertArrayAlmostEqual(subcube.data, self.expected_data)
+
+    def test_unmatched_realizations(self):
+        """Test error if realizations on temperature and lapse rate do not
+        match"""
+        temp_3d = add_coordinate(self.temperature, [0, 1, 2], 'realization')
+        lrt_3d = add_coordinate(self.lapse_rate, [2, 3, 4], 'realization')
+        msg = 'Lapse rate cube coordinate "realization" does not match '
+        with self.assertRaisesRegex(ValueError, msg):
+            _ = apply_lapse_rate(
+                temp_3d, lrt_3d, self.source_orog, self.dest_orog)
+
+    def test_missing_coord(self):
+        """Test error if temperature cube has realizations but lapse rate
+        does not"""
+        temp_3d = add_coordinate(self.temperature, [0, 1, 2], 'realization')
+        msg = 'Lapse rate cube has no coordinate "realization"'
+        with self.assertRaisesRegex(ValueError, msg):
+            _ = apply_lapse_rate(
+                temp_3d, self.lapse_rate, self.source_orog, self.dest_orog)
+
+    def test_spatial_mismatch(self):
```

Comment:
You may as well add a unit test for catching the "Destination orography projection does not match temperature grid" exception, which will likely be pretty much identical to this one.